### PR TITLE
Allow names to use a Salesforce naming convention

### PR DESCRIPTION
**Pull request by prescod**
Allow names to use a Salesforce naming convention: 

phone_number -> PhoneNumber

Alternate spellings are all valid:

phonenumber
PhoneNumber
phone_number (original)
PHONENUMBER

In the unlikely case of there being names ab_cd and abc_d, the caller can use underscores to disambiguate.

### CODE REVIEW

File: snowfakery/fakedata/fake_data_generator.py [hunk 1]
```diff
@@ -26,32 +26,41 @@ def RealisticMaybeRealEmail(self):
 
 
 class FakeData:
-    """Wrapper Faker which adds Salesforce names and case insensitivity."""
+    """Wrapper for Faker which adds Salesforce names and case insensitivity."""
 
     def __init__(self, faker: Faker):
-        self.faker = faker
-        self.fake_names = FakeNames(faker)
+        faker = faker
```
Comment (davisagli) [inline snowfakery/fakedata/fake_data_generator.py:32]: ?

File: snowfakery/fakedata/fake_data_generator.py [hunk 2]
```diff
@@ -26,32 +26,41 @@ def RealisticMaybeRealEmail(self):
...
+        fake_names = FakeNames(faker)
 
-        # canonical form of names is lower-case
-        self._lowers = {
-            v.lower(): v for v in dir(self.fake_names) if not v.startswith("_")
+        def obj_to_func_list(obj: object, canonicalizer: T.Callable):
+            return {
+                canonicalizer(name): getattr(obj, name)
+                for name in dir(obj)
+                if not name.startswith("_") and name != "seed"
```
Comment (davisagli) [inline snowfakery/fakedata/fake_data_generator.py:39]: What is "seed"?

File: snowfakery/fakedata/fake_data_generator.py (hunk 1 above)
Comment (prescod) [reply snowfakery/fakedata/fake_data_generator.py:32]: That was "self.faker = faker" and then I did a search and replace to get rid of "self.faker".

File: snowfakery/fakedata/fake_data_generator.py (hunk 2 above)
Comment (prescod) [reply snowfakery/fakedata/fake_data_generator.py]: Is it clearer in 975d5f2?

Comment (davisagli) [reply snowfakery/fakedata/fake_data_generator.py]: Not really. I knew we were ignoring "seed" but not why. Now it doesn't mention "seed" at all and I'm not sure if that's an oversight in the refactoring or what.

Comment (prescod) [reply snowfakery/fakedata/fake_data_generator.py]: Okay trying to clarify again: fc91d7e

**Review by davisagli — APPROVED** [review]

**Pull request merged**